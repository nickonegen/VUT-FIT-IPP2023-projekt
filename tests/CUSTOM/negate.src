# this program reads an integer from stdin and writes its negative to stdout
# and has many comments

.IPPcode23 # this is a header

DEFVAR GF@a
DEFVAR GF@b

# input
READ GF@a int
PUSHS GF@a

# define constants
DEFVAR GF@negative_one
MOVE GF@negative_one int@-1

# math-magic
POPS GF@a
MUL GF@b GF@a GF@negative_one

# output
WRITE GF@b