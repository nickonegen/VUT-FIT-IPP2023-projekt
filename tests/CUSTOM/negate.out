<?xml version="1.0" encoding="UTF-8"?>
<program language="IPPcode23">
	<instruction order="1" opcode="DEFVAR">
		<arg1 type="var">GF@a</arg1>
	</instruction>
	<instruction order="2" opcode="DEFVAR">
		<arg1 type="var">GF@b</arg1>
	</instruction>
	<instruction order="3" opcode="READ">
		<arg1 type="var">GF@a</arg1>
		<arg2 type="type">int</arg2>
	</instruction>
	<instruction order="4" opcode="PUSHS">
		<arg1 type="var">GF@a</arg1>
	</instruction>
	<instruction order="5" opcode="DEFVAR">
		<arg1 type="var">GF@negative_one</arg1>
	</instruction>
	<instruction order="6" opcode="MOVE">
		<arg1 type="var">GF@negative_one</arg1>
		<arg2 type="int">-1</arg2>
	</instruction>
	<instruction order="7" opcode="POPS">
		<arg1 type="var">GF@a</arg1>
	</instruction>
	<instruction order="8" opcode="MUL">
		<arg1 type="var">GF@b</arg1>
		<arg2 type="var">GF@a</arg2>
		<arg3 type="var">GF@negative_one</arg3>
	</instruction>
	<instruction order="9" opcode="WRITE">
		<arg1 type="var">GF@b</arg1>
	</instruction>
</program>